sympy/functions/elementary/integers.py

```diff
@@ -587,6 +587,53 @@ def _value_one_or_more(self, other):
             if other.is_integer and other.is_positive:
                 return S.true
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+
+        if arg0.is_finite:
+            if r.is_zero:
+                if cdir == 0:
+                    ndirl = arg.dir(x, cdir=-1)
+                    ndir = arg.dir(x, cdir=1)
+                    if ndir != ndirl:
+                        raise ValueError("Two sided limit of %s around 0"
+                                            "does not exist" % self)
+                    else:
+                        return r
```

Comment:
What if ndir is negative?

Reply:
Could you give an example of this case ?

Reply:
The leading term of `frac(x)` should be `x` if `cdir` is nonnegative and `1` if it is negative.

Reply:
For the case where we know `r` is `0`, won't 1 and 0 be the leading terms on both sides ? Maybe something like this might suffice I guess.
```
    def _eval_as_leading_term(self, x, logx=None, cdir=0):
        from sympy.calculus.accumulationbounds import AccumBounds
        arg = self.args[0]
        arg0 = arg.subs(x, 0)
        r = self.subs(x, 0)

        if arg0.is_finite:
            if r.is_zero:
                ndir = arg.dir(x, cdir=cdir)
                return S.One if ndir.is_negative else S.Zero
            else:
                return r
        elif arg0 in (S.ComplexInfinity, S.Infinity, S.NegativeInfinity):
            return AccumBounds(0, 1)
```